mod_ci/controllers.py

```diff
@@ -461,6 +461,33 @@ def progress_reporter(test_id, token):
                     g.github['repository'])
                 # If status is complete, remove the Kvm entry
                 if status in [TestStatus.completed, TestStatus.canceled]:
+                    u1 = GeneralData.query.filter(
+                        GeneralData.key == 'average_time').first()
+                    average_time = 0
+                    if u1 == None:
+                        last_running_test_id_all = TestResult.all()    
+                        if last_running_test_id_all != None:
+                            for last_running_test_id in last_running_test_id_all:
```

Comment:
This entire loop is still not very efficient. You fetch all test results, then select the test of that (through relationship), then fetch for each test all the progress data. If you go back to SQL, how would you make that a bit more efficient?